### PR TITLE
change template file name and add HCP service principal variables 

### PR DESCRIPTION
**Pull request by david-yu**
Was not able to use this learn guide without these changes.

### CODE REVIEW

File: providers.tf [hunk 1]
```diff
@@ -21,4 +21,10 @@ provider "consul" {
...
   token      = hcp_consul_cluster_root_token.token.secret_id
-}
\ No newline at end of file
+}
+
+# Replace client_id and client_secret from values from HCP IAM settings 
+provider "hcp" {
+  client_id     = "service-principal-key-client-id"
+  client_secret = "service-principal-key-client-secret"
+}
```
Comment (im2nguyen) [inline providers.tf]: ill update the tutorial to export the HCP environment variables! this is the preferred way to set provider credentials

File: windows.tf [hunk 2]
```diff
@@ -69,7 +69,7 @@ resource "aws_instance" "fakeservice" {
   vpc_security_group_ids = [module.aws_hcp_consul.security_group_id, aws_security_group.hcp_consul_ec2.id]
   key_name               = aws_key_pair.instance_key_pair.key_name
   get_password_data      = true
-  user_data = templatefile("${path.module}/templates/consul-client-agent.tpl",
+  user_data = templatefile("${path.module}/templates/consul-client-agent.tftpl",
```
Comment (im2nguyen) [inline windows.tf:72]: good catch!

**Review by im2nguyen — COMMENTED** [review]
can you remove the update to the `providers.tf`? i'll update the tutorial to use environment variables instead (recommended method). 

edit: the tutorial states that you need configure the HCP credentials for terraform, but ill add a section to make this even more explicit

image

File: providers.tf (hunk 1 above)
Comment (david-yu) [reply providers.tf]: Ok will remove thanks!

**Review by im2nguyen — APPROVED** [review]

**Pull request merged**